src/classes/OPP_StageMappingUtil.cls

```diff
@@ -91,10 +95,198 @@ public with sharing class OPP_StageMappingUtil {
+                    oppStateStages.put(oppState, stages);
+                }
+            }
+            
+            COMMITTED_STATE_SET = oppStateStages.containsKey(COMMITTED_STATE_VAL) ? oppStateStages.get(COMMITTED_STATE_VAL) : new Set<String>();
+            FINALIZED_STATE_SET = oppStateStages.containsKey(FINALIZED_STATE_VAL) ? oppStateStages.get(FINALIZED_STATE_VAL) : new Set<String>();
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Returns a list of Opportunity stage to state mappings. If it is a test running,
+    * return an artificially created list. Otherwise, get it from the database
+    ********************************************************************************************************/
+    private static List<Opportunity_Stage_To_State_Mapping__mdt> getOpportunityStageToStateMappings() {
+        List<Opportunity_Stage_To_State_Mapping__mdt> metadataRecordList = new List<Opportunity_Stage_To_State_Mapping__mdt>();
+
+        if (Test.isRunningTest()) {
+            
+            // If it's a test running, return artificial records. This ensures consistency in the testing. 
+            Opportunity_Stage_To_State_Mapping__mdt finalizedRecord = new Opportunity_Stage_To_State_Mapping__mdt();
+            finalizedRecord.Opportunity_Stage__c = 'Closed Won';
+            finalizedRecord.Opportunity_State__c = FINALIZED_STATE_VAL;
+
+            Opportunity_Stage_To_State_Mapping__mdt pledgedRecord = new Opportunity_Stage_To_State_Mapping__mdt();
+            pledgedRecord.Opportunity_Stage__c = 'Pledged';
+            pledgedRecord.Opportunity_State__c = COMMITTED_STATE_VAL;
+
+            metadataRecordList.add(finalizedRecord);
+            metadataRecordList.add(pledgedRecord);
+        } else {
+            metadataRecordList = [Select Id, Opportunity_Stage__c, Opportunity_State__c FROM Opportunity_Stage_To_State_Mapping__mdt];
+        }
+
+        return metadataRecordList;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates the finalized state record if there are no finalized records. It utilizes the 
+    * "Opportunity Stage when Fully Paid" field as the Opportunity stage for the finalized state. 
+    ********************************************************************************************************/
+    public static List<Metadata.CustomMetadata> createFinalizedOppStageToStateRecord(List<Metadata.CustomMetadata> metadataToInsert) {
+        npe01__Contacts_And_Orgs_Settings__c contactAndOrgCustomSetting = UTIL_CustomSEttingsFacade.getOrgContactsSettings();
+
+        // Only create the finalized opp stage to state record if there's no existing finalized record. 
+        Integer finalizedCount = [Select Count() From Opportunity_Stage_To_State_Mapping__mdt Where Opportunity_State__c =: FINALIZED_STATE_VAL];
```

Comment:
Same here. Upper case `SELECT`.

Reply:
I've also noticed that SOQL against the CMT object is slower than with SObjects. Definitely something to keep in mind if you're doing any queries against this in a loop.

Reply:
We cache it in the Stage to State API for that reason.  So only the first call is a cost, but it is a great callout.